src/features/Team/JoinCreateTeam.tsx

```diff
@@ -1,153 +1,136 @@
-            </Button>
-          </Box>
-          <Box>
-            <Button
-              type="button"
-              onClick={this.onClickFactory(1, fp.submitForm)}
-              isLoading={this.state.isLoading}
-              disabled={this.state.isLoading}
-            >
-              Join team
-            </Button>
-          </Box>
-        </Flex>
+  return (
+    <>
+      <div className="create">
+        <Button
+          variant={ButtonVariant.Primary}
+          isLoading={isLoading}
+          disabled={isLoading}
+          onClick={createTeam}
+        >
+          Create new team
+        </Button>
+      </div>
+
+      <Form onSubmit={(e) => { e.preventDefault(); joinTeam(); }}>
+        <div className="join">
+          <Label style={{ maxWidth: '272px' }}>
+            Already have a team?
+            <Input
+              type="text"
+              name="name"
+              value={teamCode}
+              onChange={(e) => setTeamCode(e.target.value)}
+              placeholder="Enter team code"
+              style={{ marginBottom: 0 }}
+            />
+          </Label>
+
+          <Button
+            type="submit"
+            variant={ButtonVariant.Secondary}
+            isLoading={isLoading}
+            disabled={isLoading}
+          >
+            Join team
+          </Button>
+        </div>
       </Form>
-    );
-  }
 
-  private onClickFactory(
-    submissionBtn: number,
-    submitForm: () => void
-  ): (e: any) => void {
-    return (e) => {
-      this.setState({ submissionBtn });
-      submitForm();
-    };
-  }
+      <style jsx>{`
+        .create {
+          text-align: center;
 
-  private async handleSubmit(values: FormikValues) {
-    this.setState({ isLoading: true });
-    if (this.state.submissionBtn === 0) {
-      try {
-        await Team.create({
-          name: values.name,
-          members: [this.props.hacker.id],
-        });
-        this.props.onTeamChange();
-      } catch (e) {
-        if (e.status === 409) {
-          if (e && e.data) {
-            ValidationErrorGenerator(e.data);
-          }
+          padding-top: 8px;
+          padding-bottom: 48px;
+          border-bottom: 1px solid ${theme.colors.purpleLight};
+          margin-bottom: 32px;
         }
-      }
-    } else {
-      try {
-        await Team.join(values.name);
-        this.props.onTeamChange();
-      } catch (e) {
-        if (e && e.data) {
-          ValidationErrorGenerator(e.data);
+
+        .join {
+          display: flex;
+          justify-content: space-between;
+          align-items: flex-end;
         }
-      }
-    }
-    this.setState({ isLoading: false });
+      `}</style>
+    </>
+  );
+}
+
+// Create a new random n-long character alphanumeric team code.
+// Since there are 1.24017694e25 team names possible when n=14 and
+// penalty for collision is minimal, don't have to worry
+// about handling that
+function generateRandomTeamCode(n: number): string {
```

Comment:
You could make this a server-side function in case the team code does indeed collide. What happens on API-side when there's a name collision? Fails the team create request?